### PR TITLE
[BEAM-6452] Fix NullPointerException caused by nested collections

### PR DESCRIPTION
**Pull request by reuvenlax**
A schema of the form List<List<Row>> causes a NullPointerException when converting to a POJO. This is because the Java type information was not properly passed  down recursively.

R: @kanterov

### CODE REVIEW

**Comment by apilloud** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by reuvenlax** [comment]
The examples are failing because the beam1 Jenkins executor is hosed.

**Comment by reuvenlax** [comment]
Run Java_Examples_Dataflow PreCommit

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldValueTypeInformation.java [hunk 1]
```diff
@@ -157,68 +157,94 @@ public FieldValueTypeInformation withName(String name) {
...
+        .setNullable(false)
+        .setType(mapType)
+        .setRawType(mapType.getRawType())
+        .setElementType(getArrayComponentType(mapType))
+        .setMapKeyType(getMapKeyType(mapType))
+        .setMapValueType(getMapValueType(mapType))
+        .build();
   }
-}
+}
```
Comment (apilloud) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldValueTypeInformation.java]: Nit: Add back the newline.

**Comment by apilloud** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by reuvenlax** [comment]
Run Java_Examples_Dataflow PreCommit

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldValueTypeInformation.java (hunk 1 above)
Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldValueTypeInformation.java]: done

**Comment by kanterov** [comment]
I checked the code, it looks good.

**Pull request merged**